src/libraries/System.Text.RegularExpressions/tests/RegexSRMTests.cs

```diff
@@ -465,5 +472,159 @@ public void SRMTest_ComplementFindMatch()
             Assert.Equal(6, match.Index);
             Assert.Equal(7, match.Length);
         }
+
+        [Fact]
+        public void PasswordSearch()
+        {
+            var twoLower = ".*[a-z].*[a-z].*";
+            var twoUpper = ".*[A-Z].*[A-Z].*";
+            var threeDigits = ".*[0-9].*[0-9].*[0-9].*";
+            var oneSpecial = @".*[\x21-\x2F\x3A-\x40\x5B-x60\x7B-\x7E].*";
+            var noCountUp = Not(".*(012|123|234|345|456|567|678|789).*");
+            var noCountDown = Not(".*(987|876|765|654|543|432|321|210).*");
+            // Observe that the space character (immediately before '!' in ASCII) is excluded
+            var length = "[!-~]{8,12}";
+ 
+           // Just to make the chance that the randomly generated part actually has a match
+           // be astronomically unlikely require 'P' and 'r' to be present also,
+           // although this constraint is really bogus from password constraints point of view
+            var contains_first_P_and_then_r = ".*P.*r.*";
+
+            // Conjunction of all the above constraints
+            var all = And(twoLower, twoUpper, threeDigits, oneSpecial, noCountUp, noCountDown, length, contains_first_P_and_then_r);
+
+            // search for the password in a context surrounded by word boundaries
+            var re = new Regex($@"\b{all}\b", RegexOptions.NonBacktracking | RegexOptions.Singleline);
+
+            // Does not qualify because of 123 and connot end between 2 and 3 because of \b
+            var almostPassw1 = "P@ssW0rd123";
+            // Does not have at least two uppercase
+            var almostPassw2 = "P@55w0rd";
+
+            // These two qualify
+            var password1 = "P@55W0rd";
+            var password2 = "Pa5$w00rD";
```

Comment:
That goes for tests as well :)

Reply:
Ok, will fix. I'll try to keep this in mind.